module/purpose/PS3Walk/src/PS3Walk.cpp

```diff
@@ -72,62 +83,146 @@ namespace module::purpose {
+                        case AXIS_RIGHT_JOYSTICK_HORIZONTAL:
+                            head_yaw = static_cast<double>(-event.value) / 32767.0;
                             break;
-                        case AXIS_RIGHT_JOYSTICK_VERTICAL: headPitch = static_cast<float>(-event.value); break;
-                        case AXIS_RIGHT_JOYSTICK_HORIZONTAL: headYaw = static_cast<float>(-event.value); break;
                     }
                 }
                 else if (event.isButton()) {
                     // event was a button event
                     switch (event.number) {
-                        case BUTTON_TRIANGLE:
+                        case BUTTON_START:
                             if (event.value > 0) {  // button down
                                 if (moving) {
-                                    NUClear::log("Stop walking");
-                                    emit<Task>(std::make_unique<Walk>(Eigen::Vector3d::Zero()), 1);
+                                    log<NUClear::INFO>("Stop walking");
+                                    emit<Task>(std::make_unique<Walk>(Eigen::Vector3d::Zero()), 2);
+                                    emit<Task>(std::unique_ptr<Walk>(nullptr));
                                 }
                                 else {
-                                    NUClear::log("Start walking");
+                                    log<NUClear::INFO>("Start walking");
                                 }
                                 moving = !moving;
                             }
                             break;
-                        case BUTTON_SQUARE:
+                        case BUTTON_SELECT:
                             if (event.value > 0) {  // button down
-                                if (headLocked) {
-                                    NUClear::log("Head unlocked");
+                                if (head_locked) {
+                                    log<NUClear::INFO>("Head unlocked");
                                 }
                                 else {
-                                    NUClear::log("Head locked");
+                                    log<NUClear::INFO>("Head locked");
+                                    emit<Task>(std::unique_ptr<Look>(nullptr));
                                 }
-                                headLocked = !headLocked;
+                                head_locked = !head_locked;
+                            }
+                            break;
+                            // dance moves here:
+                        case BUTTON_DPAD_UP:
+                            if (event.value > 0) {
+                                log<NUClear::INFO>("Do a dance move Dpad up");
+                                emit<Task>(std::unique_ptr<Look>(nullptr));
+                                emit<Task>(std::unique_ptr<Walk>(nullptr));
+                                emit<Task>(load_script<LimbsSequence>("StepClap1.yaml"), 3);
```

Comment:
```suggestion
                                emit<Task>(load_script<BodySequence>("StepClap1.yaml"), 3);
```
If the dance call has higher priority than Look and Walk, it should take over and you shouldn't need to disable them with `nullptr`. I'd change it to a BodySequence if you don't want Look to run.
Same for rest of the dance moves and the kicks.

Reply:
I assume you've put the nullptr in because of the freezing as you mentioned in another comment. I think if you want the robot to freeze after the moves, the walk should have been disabled, and therefore already removed.